kvstore: drop support for ca-file field in etcd client config

tklauser opened the pull request:
The usage of the ca-file field was removed in etcd v3.4. Support in Cilium was deprecated in 1.7 by commit 50867c809d58 ("etcd: use ca-file field from etcd option if available") with targeted removal in Cilium 1.8. Enough time has passed since then. Remove the backwards compatibility wrapper.

```release-note
Drop support for the ca-file field in etcd client config, deprecated in Cilium 1.7.
```

giorio94 commented on pkg/kvstore/etcd_debug.go:
```diff
@@ -321,6 +321,16 @@ func etcdDbgOutputIPs(ips []net.IP) string {
+}
+
 func etcdDbgRetrieveRootCAFile(cfgfile string) (certs [][]byte, err error) {
 	var yc yamlConfig
```
Something like:

```suggestion
	var yc struct {
		TrustedCAfile string `json:"trusted-ca-file"`
	}
```

should work as well, and allow to drop the full struct definition.

giorio94 reviewed (approved):
Thanks for the cleanup! Just one minor suggestion inline to further drop the `yamlConfig` definition.

tklauser committed:
kvstore: drop support for ca-file field in etcd client config

The usage of the ca-file field was removed in etcd v3.4. Support in
Cilium was deprecated in 1.7 by commit 50867c809d58 ("etcd: use ca-file
field from etcd option if available") with targeted removal in Cilium
1.8. Enough time has passed since then. Remove the backwards
compatibility wrapper.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
